File: libs/shape/src/main/kotlin/mono/shape/shape/MockShape.kt
/*
 * Copyright (c) 2023, tuanchauict
 */

package mono.shape.shape

import mono.graphics.geo.Rect
import mono.shape.serialization.AbstractSerializableShape

/**
 * A simple shape for testing purpose
 */
class MockShape(
    rect: Rect,
    parentId: String? = null
) : AbstractShape(id = null, parentId = parentId) {

    override fun toSerializableShape(isIdIncluded: Boolean): AbstractSerializableShape {
        TODO("Not yet implemented")
    }

    override var bound: Rect = rect
        set(value) = update {
            val isUpdated = field != value
            field = value
            isUpdated
        }
}

File: monosketch-svelte/src/lib/mono/shape/shape/mock-shape.ts
/*
 * Copyright (c) 2024, tuanchauict
 */

import type { Rect } from "$libs/graphics-geo/rect";
import type { AbstractSerializableShape } from "$mono/shape/serialization/shapes";
import { AbstractShape } from "$mono/shape/shape/abstract-shape";

/**
 * A simple shape for testing purpose
 */
export class MockShape extends AbstractShape {
    private boundInner: Rect;

    constructor(rect: Rect, parentId: string | null = null) {
        super(null, parentId);
        this.boundInner = rect;
    }

    toSerializableShape(_isIdIncluded: boolean): AbstractSerializableShape {
        throw new Error('Not yet implemented');
    }

    get bound(): Rect {
        return this.boundInner;
    }

    set bound(value: Rect) {
        this.update(() => {
            const isUpdated = this.boundInner !== value;
            this.boundInner = value;
            return isUpdated;
        });
    }
}